### PR TITLE
ES6 linting

### PR DESCRIPTION
**Pull request by aricart**
- changed var statements to const/let
- added dev dependency on @types/node to remove warnings from index.d.ts
- migrated istanbul to nyc coverage
- added package lock

### CODE REVIEW

**Comment by derekcollison** [comment]
What versions of Node.js does this cut off?

**Comment by aricart** [comment]
@derekcollison This cuts 4 which died a long time ago - 6 is also about to die on 4/30.

**Comment by derekcollison** [comment]
Agree, but wondering if we still have users on 4..

**Comment by aricart** [comment]
if we do, they are on their own on many fronts. Also the release history has versions that accommodate them.

**Comment by aricart** [comment]
@kozlovic - node-nats tests on travis have a 90% chance of failure always - they require babysitting. Imho most are invalid timeouts.

File: examples/notifications [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env node
+/*
+ * Copyright 2013-2018 The NATS Authors
```
Comment (derekcollison) [inline examples/notifications]: 2019?

File: lib/nats.js [hunk 2]
```diff
@@ -627,74 +625,73 @@ Client.prototype.scheduleHeartbeat = function() {
...
     if (undefined === stream) {
         return;
     }
 
-    stream.on('connect', function() {
-        if (client.pingTimer) {
-            clearTimeout(client.pingTimer);
-            delete client.pingTimer;
+    stream.on('connect', () => {
+        if (this.pingTimer) {
```
Comment (derekcollison) [inline lib/nats.js]: We sure this is safe? Maybe the new syntax makes this ok but I remember that `client=this` was a big deal back in the day.

Comment (aricart) [reply lib/nats.js]: Yes that is the whole thing with the arrow functions - the this there is the one in the outer block, which in this case is the client. It removes all sorts of ambiguities.

Comment (derekcollison) [reply lib/nats.js]: ok glad they fixed that about the language - lol. I am sure they did a long time ago.. I am slow.

File: examples/notifications (hunk 1 above)
Comment (aricart) [reply examples/notifications]: Yes need to update all the touched ones.

Comment (derekcollison) [reply examples/notifications]: Fix all of them that you touched.

**Pull request merged**